src/Cli.Tests/EnvironmentTests.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Cli.Tests;
+
+/// <summary>
+/// Contains test involving environment variables.
+/// </summary>
+[TestClass]
+public class EnvironmentTests
+{
+    /// <summary>
+    /// Test to verify that environment variable setup in the system is picked up correctly
+    /// when no .env file is present.
+    /// </summary>
+    [TestMethod]
+    public void TestEnvironmentVariableIsConsumedCorrectly()
+    {
+        string envVariableName = "DAB_TEST_ENVIRONMENT";
+        string jsonWithEnvVariable = @"{""envValue"": ""@env('DAB_TEST_ENVIRONMENT')""}";
+
+        // No environment File, No environment variable set in the system
+        Assert.AreEqual(null, Environment.GetEnvironmentVariable(envVariableName));
+
+        // Configuring environment variable in the system
+        Environment.SetEnvironmentVariable(envVariableName, "TEST");
+
+        // Test environment variable is correctly resolved in the config file
+        string? resolvedJson = RuntimeConfigPath.ParseConfigJsonAndReplaceEnvVariables(jsonWithEnvVariable);
+        Assert.IsNotNull(resolvedJson);
+        Assert.IsTrue(JToken.DeepEquals(JObject.Parse(@"{""envValue"": ""TEST""}"), JObject.Parse(resolvedJson)));
+
+        // removing Environment variable from the System
+        Environment.SetEnvironmentVariable(envVariableName, null);
+        Assert.ThrowsException<DataApiBuilderException>(() =>
+            RuntimeConfigPath.ParseConfigJsonAndReplaceEnvVariables(jsonWithEnvVariable),
+            $"Environmental Variable, {envVariableName}, not found.");
```

Comment:
I'd create a separate test for this scenario because if an earlier `Assert` fails then you won't get this far, but it could also be failing for different reasons, which you won't know about.